src/main/java/io/javalin/plugin/openapi/dsl/extractDocumentation.kt

```diff
@@ -56,66 +61,95 @@ private fun HandlerMetaInfo.extractDocumentationWithPathScanning(options: Create
     }
 }
 
-private fun HandlerMetaInfo.getOpenApiAnnotationFromReference(): OpenApi? {
+private fun HandlerMetaInfo.getOpenApiAnnotationFromReference(): List<OpenApi> {
```

Comment:
This becomes a list, since we include the supertype implementing the method the first time. The "in between" types where omitted, but with a recursion it is absolutely possible to reflect them too.
Children override fields of the Documentation of the supertypes